### PR TITLE
Added text-decoration challenge

### LINKED ISSUE #13468: [beta] Improve challenge "Applied Visual Design: Adjust the Hover State of an Anchor Tag"
Challenge [adjust-the-hover-state-of-an-anchor-tag](http://beta.freecodecamp.com/en/challenges/applied-visual-design/adjust-the-hover-state-of-an-anchor-tag) has an issue.

I think this challenge should be changed, I'll list some reasons:
- Right now the challenge shows two new concepts, hover and text-decoration: none, when the overall teaching style is one concept per challenge
- Here pseudo-classes are introduced for the first time in the curriculum, I think the description is too concise and could leave a beginner with a lot of doubts.
- There is not an example showing how to use :hover with a selector.

My suggestion:
- Split the challenge in two, one for hover, one for text-transform: none.
- Enhance the description, including a practical example.

### PR DESCRIPTION
**Pull request by adrianwithah**
#### Pre-Submission Checklist

- [X] Your pull request targets the `staging` branch of freeCodeCamp.
- [X] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [X] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [X] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [X] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [X] Tested changes locally.
- [X] Closes currently open issue (replace XXXX with an issue no): Closes #13468 

#### Description

Added text-decoration as a seperate challenge from the hover state challenge. Based challenge style on the "Adjust the Color of an Anchor Tag" challenge. Challenge is placed before the hover state challenge as code is to be added to the <code>a</code> selector instead of the <code>a:hover</code> selector. Also updated hover state challenge seed to transition smoothly from the text-decoration challenge.

### CODE REVIEW

File: seed/challenges/01-responsive-web-design/applied-visual-design.json [hunk 1]
```diff
@@ -1158,6 +1158,35 @@
       "translations": {}
     },
     {
+      "id": "587d781d367417b2b2517d0a",
+      "title": "Adjust the Text Format of an Anchor Tag",
```
Comment (systimotic) [inline seed/challenges/01-responsive-web-design/applied-visual-design.json]: `Text Format` is perhaps too generic for a property that only has to do with lines around the text. I'm not certain what a good title would be. Perhaps `Adjust the Text Underlining of an Anchor Tag`?

**Review by systimotic — COMMENTED** [review]
Nice work on this challenge! It sticks to the conventions set by the challenges around it really well 👍 I have a few questions and ideas.

I'm wondering a little bit about whether we should teach removing the underline from a link. This isn't great for accessibility. Perhaps we should teach something else instead? Maybe changing the colour of the underline, or underlining a header?

File: seed/challenges/01-responsive-web-design/applied-visual-design.json [hunk 2]
```diff
@@ -1158,6 +1158,35 @@
...
+        "</style>",
+        "<a href=\"http://freecatphotoapp.com/\" target=\"_blank\">CatPhotoApp</a>"
+      ],
+      "tests": [
+        "assert($('a').css('text-decoration') == 'none', 'message: The text in the anchor tag should not be underlined.');"
+      ],
+      "solutions": [],
+      "hints": [],
+      "type": "waypoint",
+      "releasedOn": "Feb 17, 2017",
```
Comment (systimotic) [inline seed/challenges/01-responsive-web-design/applied-visual-design.json]: Should this be the date, or should this get the current date? (I don't know what it should be 😅)

**Review by systimotic — COMMENTED** [review]
Forgot to click the button to add this to the previous review 😅

**Comment by hay-dee** [comment]
@systimotic Maybe i could tweak the challenge such that the link has <code>text-decoration:none</code> by default, but has an underline on hover? This design is much more prevalent in web pages and still sticks to introducing the concept of <code>text-decoration</code>.

Regarding the "releasedOn" field, i forgot to ask whether we need to change it before submitting the PR 😅 . Will have to check with other moderators on gitter.

**Comment by systimotic** [comment]
@amberbranson3 Hi! I'm sorry, I'm afraid I don't understand what you're asking. Can you elaborate? If you were looking for coding help, check out [our chatrooms](http://forum.freecodecamp.com/t/free-code-camp-official-chat-rooms/19390).

@hay-dee Sorry for the delay! I've been really busy, and will be really busy for the coming months. 
`Text Decoration` for the title sounds good to me 👍 It might not make as much sense of you don't know about the property yet, but it's probably the best option overall.
The date from the flexbox challenges is the same as the date of this challenge, so that's fine as-is.

I'm still not sure about where to use the property. Showing the underline on hover only is definitely a more common pattern, this is what GitHub does too. However, I think it would make this challenge more complicated than it should be. Maybe we should change it to a double underline?
Maybe the description should touch a bit more on the possible values?

**Comment by hay-dee** [comment]
@systimotic No problems with the delay! Thanks for taking time out to guide me on this PR. 😄 

I realised that the current text-decoration challenge is immediately **before** the [hover state challenge](http://beta.freecodecamp.com/en/challenges/applied-visual-design/adjust-the-hover-state-of-an-anchor-tag). Would swapping the order of the text-decoration and hover state challenge reduce the complexity of my proposed text-decoration challenge? Campers will then only be required to add in a line <code>text-decoration: none</code> within the already present hover state selector.

I could definitely add in a one or two-liner in the description to explain the text-decoration property and address the other possible values.

File: seed/challenges/01-responsive-web-design/applied-visual-design.json (hunk 1 above)
Comment (raisedadead) [inline seed/challenges/01-responsive-web-design/applied-visual-design.json]: IMHO, this should be `Text Styling` or `Text Decoration` with my personal preference for the later.

File: seed/challenges/01-responsive-web-design/applied-visual-design.json (hunk 2 above)
Comment (raisedadead) [inline seed/challenges/01-responsive-web-design/applied-visual-design.json]: Yes, this should be the date when we bring a new challenge for the first time into the seed.
/cc @QuincyLarson please confirm.

**Comment by raisedadead** [comment]
@hay-dee how are we coming along on this one?

**Comment by hay-dee** [comment]
Hi @raisedadead , @systimotic and i are still amidst discussion regarding the content of the challenge. Once we have come to a consensus, i can make the necessary amendments and amend this pull request :smile: .

**Comment by raisedadead** [comment]
Closing as stale. @hay-dee please re-open when ready. Thanks for your hard work and contribution.

**Comment by systimotic** [comment]
@hay-dee I'm so sorry for disappearing! As I mentioned, I was and would be busy. That turned into basically not being on GitHub from mid-March until last week. I did pass my exams, and I'm very happy I finally have time to write code again!

The proposal in your last comment sounds great to me! Do you want to pick this back up again?

**Comment by MarcDearden** [comment]
Stop sending the emails!!!

**Comment by systimotic** [comment]
Hi @MarcDearden!

It looks like you may have accidentally subscribed to all notifications for the [freeCodeCamp](https://github.com/freeCodeCamp/freeCodeCamp) repository. To stop receiving notifications please visit the [subscriptions page](https://github.com/freeCodeCamp/freeCodeCamp/subscription) or click on the image below and select **"Not watching"**.

[image](https://github.com/FreeCodeCamp/FreeCodeCamp/subscription)


We are sorry for your inconvenience!

**Comment by hay-dee** [comment]
Hi @systimotic,

I have to admit that i have not been keeping up with work in this repo. Apologies that i may not be able to carry out the changes that i have mentioned before. Maybe we can release this change to first timers? :smile:

**Pull request closed without merging**